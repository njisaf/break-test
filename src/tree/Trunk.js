import { types } from 'mobx-state-tree';

const Trunk = types.model('Trunk', {
    failures: types.map(types.frozen())
})
.actions(self => {
    return {
        failure(error, value) {
            console.log('error! ', error, value);
            self.failures.set(error, value);
        }
    }
})

export default Trunk;