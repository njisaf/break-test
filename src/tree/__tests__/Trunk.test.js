import {
    getSnapshot,
    onSnapshot,
    types
} from 'mobx-state-tree';
import Trunk from '../Trunk';
import MapUtils from '../utils/MapUtils';

describe('Trunk', () => {
    let states = [];

    const TestModel = types.compose(Trunk, MapUtils, types.model({
        identity: types.string,
        alive: true,
    }).actions(self => ({
        kill() {
            self.alive = false;
            self.failure('Test is alive?', self.alive);
        }
    })))

    const testModel = TestModel.create({
        identity: 'testOne'
    });
  
    onSnapshot(testModel, snapshot => {
        states.push(snapshot)
    })


    it('testModel.alive can toggle to false', () => {
        console.log('testModel', getSnapshot(testModel))
        expect(testModel.alive).toBe(true);
        testModel.kill();
        expect(testModel.alive).toBe(false);
    })

    it('has the dead model added to the failures map on Trunk', () => {
        expect(testModel.failures.size).toBe(1);
        expect(testModel.failures.get('Test is alive?')).toBe(false);
    })

    it('matches snapshot', () => {
        expect(states).toMatchSnapshot();
    })
})