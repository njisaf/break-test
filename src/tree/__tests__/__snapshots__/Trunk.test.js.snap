// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trunk matches snapshot 1`] = `
Array [
  Object {
    "alive": false,
    "failures": Object {
      "Test is alive?": false,
    },
    "identity": "testOne",
  },
]
`;
